packages/backend-function/src/reference_layer.ts

```diff
@@ -0,0 +1,92 @@
+import { AmplifyUserError } from '@aws-amplify/platform-core';
+import { ILayerVersion, LayerVersion } from 'aws-cdk-lib/aws-lambda';
+import { Arn } from 'aws-cdk-lib/core';
+import { Construct } from 'constructs';
+
+/**
+ * Defines the ARN regex pattern for a layer.
+ */
+export const arnPattern = new RegExp(
+  '^(arn:(aws[a-zA-Z-]*)?:lambda:[a-z]{2}((-gov)|(-iso([a-z]?)))?-[a-z]+-\\d{1}:\\d{12}:layer:[a-zA-Z0-9-_]+:[0-9]+)|(arn:[a-zA-Z0-9-]+:lambda:::awslayer:[a-zA-Z0-9-_]+)$'
+);
```

Comment:
Can you add the source of this ARN in the documentation.

Reply:
got this regex from CFN error, but modified this now to be similar to the docs now and linked the documentation.